FEXCore/Source/Interface/IR/Passes/DeadStoreElimination.cpp

```diff
@@ -105,42 +48,14 @@ struct GPRInfo {
   uint32_t kill {0};
 };
 
-bool IsFullGPR(uint32_t Offset, uint8_t Size) {
-  if (Size != 8) {
-    return false;
-  }
-  if (Offset & 7) {
-    return false;
-  }
-
-  if (Offset < 8 || Offset >= (17 * 8)) {
-    return false;
-  }
-
-  return true;
-}
-
-bool IsGPR(uint32_t Offset) {
-
-  if (Offset < 8 || Offset >= (17 * 8)) {
-    return false;
-  }
-
-  return true;
-}
-
-uint32_t GPRBit(uint32_t Offset) {
-  if (!IsGPR(Offset)) {
-    return 0;
-  }
-
-  return 1 << ((Offset - 8) / 8);
+uint32_t GPRBit(RegisterClassType Class, uint32_t Reg) {
+  return (Class == GPRClass) ? (1 << Reg) : 0;
```

Comment:
Perhaps:

```suggestion
  return (Class == GPRClass) ? (1U << Reg) : 0;
```

Just so it's always well defined, even if bit 31 isn't pragmatically hit in practice

Reply:
Sure